gradio/demo.py

```diff
@@ -0,0 +1,274 @@
+      class_names.append(dataset_info.class_names[semantic_id])
+    else:
+      class_names.append('ignore')
+
+  plt.imshow(legend, interpolation='nearest')
+  ax.yaxis.tick_left()
+  plt.yticks(range(len(legend)), class_names, fontsize=15)
+  plt.xticks([], [])
+  ax.tick_params(width=0.0, grid_linewidth=0.0)
+  plt.grid('off')
+#   plt.show()
+  return plt
+
+def run_cmd(command):
+    try:
+        print(command)
+        call(command, shell=True)
+    except KeyboardInterrupt:
+        print("Process interrupted")
+        sys.exit(1)
+MODEL_NAME = 'max_deeplab_l_backbone_os16_axial_deeplab_cityscapes_trainfine_saved_model'   
+
+
+_MODELS = ('resnet50_os32_panoptic_deeplab_cityscapes_crowd_trainfine_saved_model',
+           'resnet50_beta_os32_panoptic_deeplab_cityscapes_trainfine_saved_model',
+           'wide_resnet41_os16_panoptic_deeplab_cityscapes_trainfine_saved_model',
+           'swidernet_sac_1_1_1_os16_panoptic_deeplab_cityscapes_trainfine_saved_model',
+           'swidernet_sac_1_1_3_os16_panoptic_deeplab_cityscapes_trainfine_saved_model',
+           'swidernet_sac_1_1_4.5_os16_panoptic_deeplab_cityscapes_trainfine_saved_model',
+           'axial_swidernet_1_1_1_os16_axial_deeplab_cityscapes_trainfine_saved_model',
+           'axial_swidernet_1_1_3_os16_axial_deeplab_cityscapes_trainfine_saved_model',
+           'axial_swidernet_1_1_4.5_os16_axial_deeplab_cityscapes_trainfine_saved_model',
+           'max_deeplab_s_backbone_os16_axial_deeplab_cityscapes_trainfine_saved_model',
+           'max_deeplab_l_backbone_os16_axial_deeplab_cityscapes_trainfine_saved_model')
+_DOWNLOAD_URL_PATTERN = 'https://storage.googleapis.com/gresearch/tf-deeplab/saved_model/%s.tar.gz'
+
+_MODEL_NAME_TO_URL_AND_DATASET = {
+    model: (_DOWNLOAD_URL_PATTERN % model, cityscapes_dataset_information())
+    for model in _MODELS
+}
+
+MODEL_URL, DATASET_INFO = _MODEL_NAME_TO_URL_AND_DATASET[MODEL_NAME]
+
+model_dir = tempfile.mkdtemp()
+
+download_path = os.path.join(model_dir, MODEL_NAME + '.gz')
+urllib.request.urlretrieve(MODEL_URL, download_path)
+
+run_cmd("tar -xzvf " + download_path + " -C " + model_dir)
+
+LOADED_MODEL = tf.saved_model.load(os.path.join(model_dir, MODEL_NAME))
+def inference(image):
+    image = image.resize(size=(256, 256))
+    im = np.array(image)
+    output = LOADED_MODEL(tf.cast(im, tf.uint8))
+    return vis_segmentation(im, output['panoptic_pred'][0], DATASET_INFO)
+
+title = "Deeplab2"
+description = "demo for Deeplab2. To use it, simply upload your image, or click one of the examples to load them. Read more at the links below."
+article = "<p style='text-align: center'><a href='https://arxiv.org/abs/2106.09748'>DeepLab2: A TensorFlow Library for Deep Labeling</a> | <a href='https://github.com/google-research/deeplab2'>Github Repo</a></p>"
+
+gr.Interface(
+    inference, 
+    [gr.inputs.Image(type="pil", label="Input")], 
+    gr.outputs.Image(type="plot", label="Output"),
```

Comment:
haven't used gradio yet, would it be possible to output multiple plots instead one? for now the output plots (input / segmentation / overlay / legend) are very small to inspect

Reply:
yes multiple plots are possible, I tried to folllow the colab notebook as how the plots were presented there

Reply:
also the image is downloadable and can be opened then to read easier

Reply:
y download images would be good. just thinking having separate plots would look better on the webui (and we dont need to show the original image again). Colab shows everything in one window just due to we dont want to execute extra code blocks to show them :)

Reply:
ok sounds good, also could the colab be updated to not show the input image as well I think the layout would show the legend on the side the two plots side by side